### ISSUE TITLE
Assign a project to myself (wrangler) so that I can track what I am working on

### ISSUE THREAD

**Issue by ofanobilbao** [ebi-ait/dcp-ingest-central#19]
As a data wrangler I want to assign myself to a project I'm wrangling so that we have a clear view in the backoffice which project is being worked on by whom.

** Acceptance Criteria / Definition of Done **

* Given a wrangler is in ingest wrangler view
* When they open up a project
* Then they can add their name from a list of wranglers and save that info.

Tasks

- [x] UI design for wrangler view and ability to assign
- [ ] UI for visualisation in the project list
- [ ] Ability to filter by wranglers
- [ ] Database admin for wrangler names - dev

**Comment by gabsie** [comment]
Hey @yusra-haider 

here is my current design (however only look at the field 'primary wrangler', because that is the only thing for this sprint.

So we need an extra tab under Project, which is called Admin, and which is only visible to Wrangler access users.
Let me know if any questions
![Sprint 12 - Wrangler admin area.png](https://images.zenhubusercontent.com/5b18f77c348b7966b4345c29/fdd1ed13-a1c4-4f41-91ef-92871ee31600)
Thanks!
Gabby

**Comment by yusra-haider** [comment]
Thanks for the designs @gabsie!

**Comment by yusra-haider** [comment]
Comments from the demo for this feature:

- can the wranglers see the projects that they're assigned as Primary Wranglers to?
- @MightyAx confirmed that the `My Projects` view shows both projects created by the current user as well as projects that are meant to be wrangled by the current user
- todo: check what the view looks like 

cc: @clairerye  @gabsie

**Comment by clairerye** [comment]
waiting on PR

**Comment by gabsie** [comment]
Hi, @yusra-haider 

This looks good as a start and functionality.
1. Can we, when clicking Edit on the section we're on (Admin area) just go to this specific section to edit?
2. Also, this list needs cleaning (removing the people who have left the team  - Justin, Rodrey, Rolando, etc)
3. We seem to have lost the top navigation once in Edit mode (we don't have the step by step flow of Project/Experiment etc)
4. I thought you wanted to have this on the top level (not under the project tabs, what made you change your mind?) :)


I will create a new ticket for myself which is to  make sure the designs look more like what I have in Figma, for the whole portal.

Thanks for now!!! :)
Gabs

**Comment by yusra-haider** [comment]
Thanks for the feedback @gabsie 

For: 

1. That's how the `edit` button currently works for all tabs, Contacts, Publications, Funders etc.
    Do we want to change how that works for all these tabs or just for the Admin area?

2. yes, the list is dynamic and gets all the wranglers from the ingest database. We need to eventually have some database cleanups.

3. I am not sure how the top navigation used to look in Edit mode, but it has been like that for a while now. It's not something introduced by the addition of the Admin tab. The top navigation bar comes back when you're out of Edit mode. 

4. I thought it made more sense to follow the current design as I agree with the bit where you mentioned that the top bars were like progress bars. The Admin bar wouldn't fit with as a progress bar

**Comment by mshadbolt** [comment]
Rough testing plan from a wrangler perspective based on some ideas from @Wkt8 and @yusra-haider:

Testing on dev:

### Basic functionality: 
- [ ] the primary wrangler field is editable and behaves as expected in edit mode
- [ ] the admin tab isn't visible at all for non-wranglers

### Interacting as expected:

#### If I go to an existing project
- [ ] Projects that I wrangled seem to already have me as primary wrangler (checked a couple)
- [ ] I can see the admin area and change the primary wrangler in edit mode
- [ ] the primary wrangler I change it to can see the project in their 'My Projects' list

#### If I create a project in the ui:
- [ ] I can see the project in ‘My Projects’
- [ ] I am assigned primary wrangler by default but am able to change this 
What happens if I assign it to someone else?
a. Can I see it in ‘My Projects’?
b. Can they see it in their ‘My Projects’ 

#### If I create a project via spreadsheet:
If I assign it to me using the admin tab: 
- [ ] I can see it in ‘My Projects’
If I assign it to someone else:
- [ ] can they see it in their 'My Projects'

General feedback:
- We probably need to periodically review the wrangler list, there are several users who should no longer have wrangler rights e.g., Justin, Rolando, Rodrey, 'Mark' Goite.

**Comment by gabsie** [comment]
Hi, both
Yes, next steps and one of the tickets for me to work on is the filtering by wrangler.
Yusra, I am good for this to go in prod.
Did we at least clean up the list of people? For now let's remove Rodrey/Rolando/Justin, as well as my name appears twice?

Thanks!
Gabs

**Comment by gabsie** [comment]
I created a ticket - 312
